src/ray/id.cc

```diff
@@ -160,51 +94,35 @@ size_t UniqueID::hash() const {
   return hash_;
 }
 
-std::ostream &operator<<(std::ostream &os, const UniqueID &id) {
-  if (id.is_nil()) {
-    os << "NIL_ID";
-  } else {
-    os << id.hex();
+size_t ObjectID::hash() const {
+  // Note(ashione): hash code lazy calculation(it's invoked every time if hash code is
+  // default value 0)
+  if (!hash_) {
+    hash_ = MurmurHash64A(this, ObjectID::size(), 0);
   }
-  return os;
-}
-
-const ObjectID ComputeObjectId(const TaskID &task_id, int64_t object_index) {
-  RAY_CHECK(object_index <= kMaxTaskReturns && object_index >= -kMaxTaskPuts);
-  ObjectID return_id = ObjectID(task_id);
-  int64_t *first_bytes = reinterpret_cast<int64_t *>(&return_id);
-  // Zero out the lowest kObjectIdIndexSize bits of the first byte of the
-  // object ID.
-  uint64_t bitmask = static_cast<uint64_t>(-1) << kObjectIdIndexSize;
-  *first_bytes = *first_bytes & (bitmask);
-  // OR the first byte of the object ID with the return index.
-  *first_bytes = *first_bytes | (object_index & ~bitmask);
-  return return_id;
+  return hash_;
 }
 
-const TaskID FinishTaskId(const TaskID &task_id) {
-  return TaskID(ComputeObjectId(task_id, 0));
-}
+size_t TaskID::hash() const { return MurmurHash64A(this, TaskID::size(), 0); }
```

Comment:
Why not cache hash code for TaskID?

Reply:
Now TaskID is nested in ObjectID, if TaskID has a value of cache_, Objected cannot contain it with one index field. In the future, the TaskID will have a int32_t index which can be used as a hash.